### PR TITLE
Added file content view.

### PR DESCRIPTION
**Pull request by enyedi-ervin-30126**

### CODE REVIEW

File: src/main/java/utcn/repoviewer/FileView.java [hunk 1]
```diff
@@ -29,14 +32,53 @@ public FileView(String studentName, String pathToFileAbsolute){
...
+        this.textAreaCode = new JTextArea(fileContent);
         textAreaCode.setAlignmentX( Component.CENTER_ALIGNMENT );
+        if (fileContent.isEmpty()){
+            textAreaCode.setEnabled(false);
+        }
         this.add(textAreaCode);
     }
     
+    public String modifyPath(String absolutePath){
+
```
Comment (cuibus) [inline src/main/java/utcn/repoviewer/FileView.java:45]: code cosmetics: delete empty line here

File: src/main/java/utcn/repoviewer/FileView.java [hunk 2]
```diff
@@ -29,14 +32,53 @@ public FileView(String studentName, String pathToFileAbsolute){
...
+        String fileContent = getFileContents(pathToFileAbsolute);
+        this.textAreaCode = new JTextArea(fileContent);
         textAreaCode.setAlignmentX( Component.CENTER_ALIGNMENT );
+        if (fileContent.isEmpty()){
+            textAreaCode.setEnabled(false);
+        }
         this.add(textAreaCode);
     }
     
+    public String modifyPath(String absolutePath){
```
Comment (cuibus) [inline src/main/java/utcn/repoviewer/FileView.java:44]: Why do we need this method anyway ? The path of the absolute file should be ready when the FileView object is created. While debugging, I discovered that the path received from Main looks like this:
image
This is a bug which might have been introduced by a member of the team, but you must fix this before adding unnecessary code in the FileView.
I'd say there are 2 options:
1. You either get the absolute path of the file directly from Main (it is fully computed there)
2. You get base path, student name, and relative path of the file and you compute the absolute path here.

**Review by cuibus — CHANGES_REQUESTED** [review]

File: src/main/java/utcn/repoviewer/FileView.java (hunk 2 above)
Comment (enyedi-ervin-30126) [reply src/main/java/utcn/repoviewer/FileView.java:44]: I discovered that the incorrect path is returned by a method which can not be modified. 
`String firstSelectedPath = treeFolders.getSelectionPath().toString();`
The getSelectionPath() returns the path incorrectly in Main. 

Should I modify the path which is sent to the FileView constructor here? I could not understand how should I implement the second option which you have suggested.

Comment (cuibus) [reply src/main/java/utcn/repoviewer/FileView.java:44]: Indeed, if you take the selection path you get a TreePath object. If you call .toString() immediately on that object, you get a string which contains three parts. I'm guessing you only need the second and third part from that object.

File: src/main/java/utcn/repoviewer/FileView.java [hunk 3]
```diff
@@ -29,14 +32,29 @@ public FileView(String studentName, String pathToFileAbsolute){
         this.add(labelStudentName);
         
         this.pathToFileAbsolute = pathToFileAbsolute;
-        this.textAreaCode = new JTextArea(getFileContents(pathToFileAbsolute));
+        String fileContent = getFileContents(pathToFileAbsolute);
+        this.textAreaCode = new JTextArea(fileContent);
         textAreaCode.setAlignmentX( Component.CENTER_ALIGNMENT );
+        if (fileContent.isEmpty()){
+            textAreaCode.setEnabled(false);
+        }
```
Comment (cuibus) [inline src/main/java/utcn/repoviewer/FileView.java:40]: Why do we disable the textAreaCode on condition ? The textAreaCode should not be editable in any case.

**Review by cuibus — CHANGES_REQUESTED** [review]
We can check the merging of this branch together, it's going to be a bit complicated because @nicorici-iulia-cristina-30126 also worked on the FileView.

File: src/main/java/utcn/repoviewer/FileView.java (hunk 3 above)
Comment (enyedi-ervin-30126) [reply src/main/java/utcn/repoviewer/FileView.java:40]: The ticked said that if there is no file with the same name, the area should be grayed out.

**Review by cuibus — APPROVED** [review]

**Review by cuibus — APPROVED** [review]

**Pull request merged**